packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,304 @@
+    }
+
+    this.logger.info(
+      'Getting audit logs (organizationId=%s, filter=%o, pagination=%o)',
+      organizationSlug,
+      filter,
+      pagination,
+    );
+    const limit = sql.raw(String(pagination?.first ?? 25));
+    const newCursorId = pagination?.cursorId ? decodeHashBasedCursor(pagination?.cursorId) : null;
+    const lastYear = this.formatToClickhouseDateTime(
+      new Date(new Date().setFullYear(new Date().getFullYear() - 1)).toISOString(),
+    );
+    const cursorTimestamp = pagination?.cursorTimestamp
+      ? this.formatToClickhouseDateTime(pagination.cursorTimestamp)
+      : lastYear;
+
+    const where: SqlValue[] = [];
+    where.push(sql`organization_id = ${organizationSlug}`);
+
+    if (filter?.startDate && filter?.endDate) {
+      const from = this.formatToClickhouseDateTime(filter.startDate.toISOString());
+      const to = this.formatToClickhouseDateTime(filter.endDate.toISOString());
+      where.push(sql`timestamp >= ${from} AND timestamp <= ${to}`);
+    }
+    if (pagination?.cursorId && pagination?.cursorTimestamp) {
+      where.push(sql`id < ${String(newCursorId?.id)}`, sql`timestamp = ${cursorTimestamp}`);
+    } else if (pagination?.cursorId) {
+      where.push(sql`id < ${String(newCursorId?.id)}`);
+    } else if (pagination?.cursorTimestamp) {
+      where.push(sql`timestamp = ${cursorTimestamp}`);
+    } else {
+      where.push(sql`timestamp <= ${cursorTimestamp}`);
+    }
+
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT *
+        FROM audit_logs
+        ${whereClause}
+        ORDER BY timestamp DESC, id DESC
+        LIMIT ${limit}
+      `,
+      queryId: 'get-audit-logs',
+      timeout: 10000,
+    });
+
+    const data = AuditLogClickhouseArrayModel.parse(result.data);
+
+    return {
+      data,
+    };
+  }
+
+  async totalAuditLogs(organizationSlug: string): Promise<number> {
+    const isOwner = await this.auth.isOwnerOfOrganization({
+      organization: organizationSlug,
+    });
+    if (!isOwner) {
+      throw new GraphQLError('Unauthorized: You are not authorized to perform this action');
+    }
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT count() as total
+        FROM audit_logs
+        WHERE organization_id = ${organizationSlug}
+      `,
+      queryId: 'total-audit-logs',
+      timeout: 10000,
+    });
+
+    const parsedResult = TotalAuditLogsSchema.safeParse(result.data[0]);
+    if (!parsedResult.success) {
+      throw new Error('Failed to parse audit log count result');
+    }
+
+    return Number(parsedResult.data.total);
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: string; endDate?: string },
+    pagination?: { first: number; cursorId: string | null; cursorTimestamp: string | null },
```

Comment:
@TuvalSimha We don't need pagination for exporting. The export is based on a start and a end date.